craft_parts/parts.py

```diff
@@ -177,12 +177,39 @@ def get_scriptlet(self, step: Step) -> Optional[str]:
         return self._data.get(scr[step])
 
 
+def part_list_by_name(names: Optional[List[str]], part_list: List[Part]) -> List[Part]:
+    """Return a list of parts from part_list that are named in names.
+
+    :param names: The list of part names. If the list is empty or not
+        defined, return all parts from part_list.
+    :param part_list: The list of all known parts.
+
+    :returns: The list of parts corresponding to the given names.
+
+    :raises InvalidPartName: if a part name is not defined.
+    """
+    if names:
+        # check if all part names are valid
+        valid_names = [p.name for p in part_list]
+        for name in names:
+            if name not in valid_names:
+                raise errors.InvalidPartName(name)
```

Comment:
It's easier to do...

```
        valid_names = {p.name for p in part_list}
        invalid_names = set(names) - valid_names
        if invalid_names:
            ...
```

I like this more because you don't end up with an `in` comparison inside a `for` loop.

Reply:
ack